java-checks/src/main/java/org/sonar/java/checks/NullPointerCheck.java

```diff
@@ -208,6 +235,14 @@ public void visitVariable(VariableTree tree) {
     }
   }
 
+  @Override
+  public void visitWhileStatement(WhileStatementTree tree) {
+    ConditionalState conditionalState = visitCondition(tree.condition());
+    currentState = conditionalState.trueState;
+    scan(tree.statement());
+    currentState = currentState.parentState.merge(conditionalState.trueState, conditionalState.falseState);
```

Comment:
aren't we gonna end up with no valuable information with a merge like this one ?